### PR TITLE
feat: add IsUserLoggedIn or IsUserLoggedOut component

### PR DESCRIPTION
**Pull request by gabriele-ct**
contributes to #1617

### CODE REVIEW

File: websites/docs-smoke-test/src/content/index.mdx [hunk 1]
```diff
@@ -29,3 +30,13 @@ title: Explore how to build a documentation website for commercetools
...
+
+# Logged in or logged out content
+
+<IsUserLoggedIn>
+ ## Welcome back!
+</IsUserLoggedIn>
+
+<IsUserLoggedOut>
+ ## Please log in
+</IsUserLoggedOut>
```
Comment (nkuehn) [inline websites/docs-smoke-test/src/content/index.mdx:42]: Would it be more readable or "template language like"  if it were `<IfUserLoggedIn>`  vs. `<IsUserLoggedIn>` ?

File: websites/docs-smoke-test/src/content/index.mdx [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ---
 title: Explore how to build a documentation website for commercetools
 ---
+import { IsUserLoggedIn, IsUserLoggedOut } from '@commercetools-docs/gatsby-theme-learning'
```
Comment (nkuehn) [inline websites/docs-smoke-test/src/content/index.mdx:4]: I think this component is small enough to be provided automatically through the shortcodes file.

File: websites/docs-smoke-test/src/content/index.mdx (hunk 1 above)
Comment (gabriele-ct) [reply websites/docs-smoke-test/src/content/index.mdx]: Yes, agreed

File: websites/docs-smoke-test/src/content/index.mdx (hunk 2 above)
Comment (gabriele-ct) [reply websites/docs-smoke-test/src/content/index.mdx]: Sure, makes sense

**Comment by nkuehn** [comment]
P.S:  to reduce the flicker time on reloads we could set the Auth0 Provider cache to localstorage: https://auth0.github.io/auth0-react/interfaces/Auth0ProviderOptions.html#cacheLocation  

This would generally improve UX when switching between microsites

**Pull request merged**